llvm/include/llvm/Analysis/DXILResource.h

```diff
@@ -11,19 +11,192 @@
 
 #include "llvm/ADT/MapVector.h"
 #include "llvm/ADT/StringRef.h"
+#include "llvm/IR/GlobalVariable.h"
 #include "llvm/IR/PassManager.h"
 #include "llvm/Pass.h"
 #include "llvm/Support/Alignment.h"
 #include "llvm/Support/DXILABI.h"
 
 namespace llvm {
 class CallInst;
+class DataLayout;
 class LLVMContext;
 class MDTuple;
+class TargetExtType;
```

Comment:
You can't derive from a forward-declared type so you must be including the header or depending on an include ordering.

Reply:
Good point. I've replaced this with an explicit include of DerivedTypes.h